src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -410,12 +435,16 @@ def extract_return_input_types(
            2. return_type and input_types are not provided, but type hints are provided,
               then just use the types inferred from type hints.
     """
-
     (
         return_type_from_type_hints,
         input_types_from_type_hints,
     ) = get_types_from_type_hints(func, object_type, output_schema)
-    if installed_pandas and return_type and return_type_from_type_hints:
+    if (
+        installed_pandas
+        and return_type
+        and return_type_from_type_hints
+        and input_types is not None
```

Comment:
This condition is new for this if-statement. Is this intended?

Reply:
This is intentional. The below code fails if input_types is None so in practice I don't think it ever is None, but mypy does not pick up on that and complains.